### PR TITLE
Adding an ambush flag for monsters

### PR DESCRIPTION
**Pull request by Interrupt**
This makes monsters quiet and still until they actually see the player.

### CODE REVIEW

File: Dungeoneer/src/com/interrupt/dungeoneer/entities/Monster.java [hunk 1]
```diff
@@ -562,8 +576,14 @@ else if(alerted && playerdist <= 1f && !fleeing) // If close enough to the playe
 			targety = player.y;
 			stuckWanderTimer = 0f;
 		}
+
+		// Wander when not alerted
+		boolean canWander = wanders && !alerted && ambushMode != AmbushMode.None;
+
+		// Wander when there is no other path to the target
+		boolean stuckWander = !canSafelyNavigateToTarget || !chasetarget || stuckWanderTimer > 0;
```
Comment (joshuaskelly) [inline Dungeoneer/src/com/interrupt/dungeoneer/entities/Monster.java:584]: I appreciate this. 👍

Comment (PythooonUser) [reply Dungeoneer/src/com/interrupt/dungeoneer/entities/Monster.java:584]: Yes, makes that part much more readable.

**Comment by PythooonUser** [comment]
Logic seems flipped: https://discord.com/channels/266998536632139776/707655151279931392/813134759521812531

**Review by joshuaskelly — APPROVED** [review]
Looks good! Thanks or fixing the logic issue! 👍

**Review by PythooonUser — APPROVED** [review]
Looks good!

**Pull request merged**